uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -12,6 +16,28 @@ function tasksassignments_civicrm_config(&$config) {
 }
 
 /**
+ * @param array $wrappers
+ * @param array $apiRequest
+ */
+function tasksassignments_civicrm_apiWrappers(&$wrappers, $apiRequest) {
+  $requestJson = ArrayHelper::value('json', $_REQUEST);
+  $params = ArrayHelper::value('params', $apiRequest, []);
+  $optionGroup = ArrayHelper::value('option_group_id', $params);
+  $isActivityType = $optionGroup === 'activity_type';
+  $entity = ArrayHelper::value('entity', $apiRequest);
+  $action = ArrayHelper::value('action', $apiRequest);
+
+  $jsonFromCaseEdit = '"actTypes":["OptionValue","get",{"option_group_id":"activity_type","options":{"sort":"name","limit":0}}]';
+  $isAssignmentEdit = strpos($requestJson, $jsonFromCaseEdit) !== FALSE;
+
+  if ($isAssignmentEdit && $entity === 'OptionValue' && $action === 'get' && $isActivityType) {
+    $wrappers[] = new StripInvalidTaskTypesWrapper();
+  } elseif ($entity === 'CaseType' && $action === 'create') {
```

Comment:
I wasn't aware of the hrrequirement CaseType usages. If CaseType is being used elsewhere in CiviHR then this changes how we should handle this for sure.

I think we definitely should have strict validation because relying on frontend validation is just weak, and can easily be bypassed. If we have business logic that requires that users are not allowed to create assignment types with anything other than "CiviTask" or "CiviDocument" then I think we should enforce that on the backend.

This is another example of the problem with hacking the case page to our use for assignments, we should really have a separate endpoint to use here where we can apply this logic. I'm not comfortable with removing this validation, but if you are sure that it will break other parts (e.g. hrrequirement) then I will remove it.